src/components/fileviewer.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+
+import * as FetchAPI from '../fetch_data';
+
+export default class FileViewerContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      appError: undefined,
+      rawFile: undefined,
+    };
+
+    const { revision, path } = this.props;
+    FetchAPI.getRawFile(revision, path)
+      .then(response => {
+        if (response.status !== 200) {
+          console.log('Error status code' + response.status);
+          return;
+        }
+        response.text().then(file => {
+          this.setState({ rawFile: file.split('\n') })
+        });
+      })
+      .catch((error) => {
+        console.error(error);
+        this.setState({
+          appError: 'We did not manage to parse the raw-file correctly.',
+        });
+      });
+  }
+
+
+  render() {
+    // console.log(this.props);
+    const { revision, path } = this.props;
+    return (
+      <div>
+        <FileViewerMeta revision={revision} path={path}/>
+        <FileViewer rawFile={this.state.rawFile} />
+      </div>
+    );
+  }
+}
+
+
+
+/* This viewer renders each line of the file with its line number */
+const FileViewer = ({ rawFile }) => {
+  if (!rawFile) {
+    return( <div>"Waiting for file from the backend"</div> );
+  }
+
+  var id = 0;
+  const rawFileLines = rawFile.map((line) => {
+    id = id + 1;
+    return(
+      <tr key={id}>
+        <th>{id}</th>
+        <th>{line}</th>
+      </tr>
+    );
+  });
+
+  return (
+    <div>
+      <table className="table table-sm table-bordered table-hover">
+        <tbody>
+          {rawFileLines}
```

Comment:
Please inline `rawFileLines`.  I will admit inlining is a judgment call. I inline anything that is used only once: We do not need to be assigning names, and increasing the cognitive load, when we do not need to. Taken to the extreme, this can lead to very big functions, which I tend to do.   Another reason for inline in this case is the caller is very simple; it has no loops, and is not too deeply indented.